src/System-Support/SystemDictionary.class.st

```diff
@@ -325,14 +324,19 @@ SystemDictionary >> removeClassNamed: aName [
 	self at: aName asSymbol ifPresent: [:oldClass | oldClass removeFromSystem]
 ]
 
+{ #category : #removing }
+SystemDictionary >> removeFromCaches: aClassName [
```

Comment:
Rename `aClassName` to `aKey`, because it's  also trying to remove it from `cachedNonClassNames`, so it looks like it's not always a class name.